### PR TITLE
Clean up scheme usage and type conversion

### PR DESCRIPTION
**Pull request by karlkfi**
- Add core.Scheme, to be used eveywhere in this code base.
  This scheme should have all required typed registered.
- Add utility functions for type lookup and conversion:
  - kinds.ToUnstructured
  - kinds.ToTypedObject
  - kinds.ToTypedWithVersion
  - kinds.ToUnstructuredWithVersion
  - kinds.ToUnstructureds
  - kinds.Lookup
- Refactor most instances of type conversion and detection to use
  the new convertion code, for consistency.
- Fix nomos hydrate output to skip the empty metadata field.
  This was a bonus from using a scheme with the List type registered.

### CODE REVIEW

File: pkg/kinds/convert_test.go [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+						"ports": []interface{}{
+							map[string]interface{}{
+								"protocol":   "TCP",
+								"port":       int64(80),   // Type change
+								"targetPort": int64(9376), // Type change
+							},
+						},
+					},
+					"status": map[string]interface{}{ // Added field
+						"loadBalancer": map[string]interface{}{}, // Added field
```
Comment (haiyanmeng) [inline pkg/kinds/convert_test.go:526]: Let us add the discussion in https://github.com/GoogleContainerTools/kpt-config-sync/pull/77#discussion_r961950738 to here as a comment to help others understand this.

Comment (karlkfi) [reply pkg/kinds/convert_test.go:539]: done

**Comment by karlkfi** [comment]
Fixed the e2e tests (i hope). There were a bunch of e2e test json artifacts i needed to update too, not just the unit test json artifacts.

And yes, this should probably be noted on release notes, since it changes nomos hydrate output (removes empty metadata field in the top-level List).

**Comment by karlkfi** [comment]
/retest

Possible flake in mono-repo TestChangeCustomResourceDefinitionSchema:
```
ERROR: status.sync.inProgress contains changes that haven't been synced
```

**Pull request merged**